### PR TITLE
test: Cypress - Flaky fix

### PR DESCRIPTION
**Pull request by Vijetha-Kaja**
## Description

- RefreshPage() method update
- Replaced External API in ApiBugs_Spec.ts

**Fixed below flaky tests**

- DSCrudAndBindings_Spec.ts
- GitSyncedApps_spec.js
- S3_1_spec.js

## Type of change

- Flaky test fix

## How Has This Been Tested?
- Cypress test runs

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Binding/TableTextPagination_spec.js [hunk 1]
```diff
@@ -138,6 +138,11 @@ describe("Test Create Api and Bind to Table widget", function () {
 
   it("5. Table-Text, Validate Server Side Pagination of Paginate with response URL", function () {
     /**Validate Response data with Table data in Text Widget */
+    cy.get("body").then(($ele) => {
+      if ($ele.find(locators._backToEditor).length) {
+        deployMode.NavigateBacktoEditor();
+      }
+    });
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Binding/TableTextPagination_spec.js:145]: Why is this needed?

Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Binding/TableTextPagination_spec.js:145]: To make it work on re-run

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Binding/TableTextPagination_spec.js:145]: But why it will fail in 1st run? Even after `cy.get("@postExecute.all");`?

Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Binding/TableTextPagination_spec.js:145]: Mismatch of data still happened in one run yesterday,  so thought of making the re-run attempt work

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Binding/TableTextPagination_spec.js:145]: No, skip this until properly fixed now - in this PR itself.

File: app/client/cypress/e2e/Regression/ServerSide/QueryPane/S3_1_spec.js [hunk 2]
```diff
@@ -457,7 +457,8 @@ describe("Validate CRUD queries for Amazon S3 along with UI flow verifications",
       cy.ClickGotIt();
 
       //Verifying Searching File from UI
-      cy.xpath(queryLocators.searchFilefield).type("S3Crud_").wait(10000); //for search to finish
+      agHelper.TypeText(queryLocators.searchFilefield, "S3Crud_{enter}");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ServerSide/QueryPane/S3_1_spec.js:460]: No revert to previous. Instead of search text `S3Crud_` - pull up few more - say 4 chars from the new guid string & search.

**Review by Aishwarya-U-R — APPROVED** [review]
